digest/digester.go

```diff
@@ -18,27 +21,44 @@ type Digester struct {
 func NewDigester(alg string, h hash.Hash) Digester {
 	return Digester{
 		alg:  alg,
-		hash: h,
+		Hash: h,
 	}
 }
 
 // NewCanonicalDigester is a convenience function to create a new Digester with
-// out default settings.
+// our default settings.
 func NewCanonicalDigester() Digester {
 	return NewDigester("sha256", sha256.New())
 }
 
-// Write data to the digester. These writes cannot fail.
-func (d *Digester) Write(p []byte) (n int, err error) {
-	return d.hash.Write(p)
-}
-
 // Digest returns the current digest for this digester.
 func (d *Digester) Digest() Digest {
-	return NewDigest(d.alg, d.hash)
+	return NewDigest(d.alg, d.Hash)
+}
+
+// ResumableDigester is a digester that can export its internal state and be
+// restored from saved state.
+type ResumableDigester struct {
+	alg string
+	crypto.ResumableHash
+}
+
+// NewResumableDigester creates a new ResumableDigester with the given hashing
+// algorithem and instance of that resumable hasher.
+func NewResumableDigester(alg string, h crypto.ResumableHash) ResumableDigester {
```

Comment:
Can we defined ResumableHash in this package and type switch to get to your custom version?  I wouldn't want to expose the custom crypto interface in our exported interface.